lib/utilities/paginator.rb

```diff
@@ -0,0 +1,106 @@
+module Utilities
+  module Paginator
+    def items_to_display(model:, items_per_page: 5, current_page: 1)
+      @items_per_page = items_per_page
+      @current_page = current_page.to_i
+      @number_of_items = model.count
+      @last_page = (@number_of_items.to_f / @items_per_page).ceil
+      model.offset(items_per_page * (@current_page - 1))
+           .limit(items_per_page)
+    end
+
+    def paginator(max_page_blocks: 7)
+      render partial: 'shared/paginator',
+             locals: { blocks: pages_to_display(@current_page,
+                                                max_page_blocks,
+                                                @last_page),
+                       current_page: @current_page,
+                       previous_link_css: previous_link_css,
+                       next_link_css: next_link_css,
+                       last_page: @last_page }
+    end
+
+    def page_item_css(page, current_page)
+      if page == current_page
+        'page-item active'
+      elsif page.to_s == '...'
+        'page-item disabled'
+      end
+    end
+
+    private
+
+    def previous_link_css
+      @current_page == 1 ? 'page-item disabled' : 'page-item'
+    end
+
+    def next_link_css
+      @current_page == @last_page ? 'page-item disabled' : 'page-item'
+    end
+
+    def pages_to_display(current_page, max_page_blocks, number_of_pages)
+      if all_pages_fit?(number_of_pages, max_page_blocks)
+        complete_pagination(number_of_pages)
+      elsif current_page_at_the_beginning?(current_page, max_page_blocks)
+        beginning_pagination(max_page_blocks)
+      elsif current_page_at_the_end?(current_page,
+                                     number_of_pages, max_page_blocks)
```

Comment:
I remember why I did it now - Rubocop won't let me have 11 lines of code, so. I shuffled it around - 2 lines on top, one at the bottom, hope it looks better.